src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -258,9 +258,24 @@ def handle_result_for_project_active_mode(
             if not self.has_reached_status_threshold(project_subscription, result["status"]):
                 return
 
-            if features.has(
-                "organizations:uptime-create-issues", project_subscription.project.organization
-            ):
+            issue_creation_flag_enabled = features.has(
+                "organizations:uptime-create-issues",
+                project_subscription.project.organization,
+            )
+
+            # Do not create uptime issue occurences for
+            restricted_host_provider_ids = options.get(
+                "uptime.restrict-issue-creation-by-hosting-provider-id"
+            )
+            issue_creation_restricted_by_provider = (
+                project_subscription.uptime_subscription.host_provider_id
+                in restricted_host_provider_ids
+            )
+
+            if issue_creation_restricted_by_provider:
+                metrics.incr("uptime.result_processor.restricted_by_provider", sample_rate=1.0)
```

Comment:
Wondering if we should include the provider here or not... I can't imagine this having more than 10 or so values. Or maybe a log with extra info if we don't want it in datadog?

Reply:
Ah yeah true. That's probably pretty low cardinality. I'll toss it in